src/brogue/Architect.c

```diff
@@ -3445,8 +3445,10 @@ void restoreMonster(creature *monst, short **mapToStairs, short **mapToPit) {
         } else {
             theMap = mapToStairs;
         }
+
+        pmap[*x][*y].flags &= ~HAS_MONSTER;
```

Comment:
That line refers to the monster's position before moving, the second `pmap[*x][*y].flags |= HAS_MONSTER` is done on the new tile.

Reply:
I see! So, are all monsters "restored" when revisiting a floor by this function? Then the need for HAS_MONSTER being a permanent flag is because off-level allies/pursuers move and may be restored prior to other monsters, causing them to superimpose and glitched out?

Reply:
Yeah, that's what happens here.